### PR TITLE
Laboratory work #2, Menyalshchikov Timur - 21FPL1 

### PR DESCRIPTION
**Pull request by Timur20033**

### CODE REVIEW

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -164,15 +164,15 @@ def calculate_tf(frequencies: dict[str, int]) -> Optional[dict[str, float]]:
 
 def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optional[dict[str, float]]:
     """
-    Calculates TF-IDF score for each of the tokens
-    based on its TF and IDF scores
+       Calculates TF-IDF score for each of the tokens
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:167]: revert this change

File: lab_2_keywords_cooccurrence/main.py [hunk 2]
```diff
@@ -17,7 +17,34 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(text, str):
+        return None
+    if text == '':
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:22]: simplify - use the falsy mechanism

File: lab_2_keywords_cooccurrence/main.py [hunk 3]
```diff
@@ -17,7 +17,34 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
...
+        return None
+    if all(symbol.isalpha() is False for symbol in text):
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:24]: `all`? Or `any`?

File: lab_2_keywords_cooccurrence/main.py [hunk 4]
```diff
@@ -17,7 +17,34 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
...
+        return []
+    level_one = text[:]
+    for symbol in level_one:
+        if symbol in ['!', '(', ')', '-', '[', ']', '{', '}', ';', ':', '"', ',', "'", '¡', '¿', '-',
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:28]: move the punctuation a variable. Or use `string.punctuation`

File: lab_2_keywords_cooccurrence/main.py [hunk 5]
```diff
@@ -17,7 +17,34 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
...
+                      '<', '>', '/', '?', '@', '#', '$', '%', '^', '&', '*', '_', '~', ':', '⟩', '⟨',
+                      '—', '–', '«', '»']:
+            level_one = level_one.replace(symbol, '.')
+    level_two = ''
+    for idx, symbol in enumerate(level_one):
+        if not (idx not in [-1, 0] and symbol == " " and level_one[idx + 1] == "."):
+            level_two += symbol
+    level_three = ''
+    for idx, symbol in enumerate(level_two):
+        if not(symbol == '.' and level_two[idx - 1] == '.'):
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:38]: revert not and just reverse the condition

File: lab_2_keywords_cooccurrence/main.py (hunk 3 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py:24]: Here I check option if all elements are punctuation. So, if all of them are symbols of punctuation, I return empty list. I don't know how I can use here 'any' instead of 'all'...

File: lab_2_keywords_cooccurrence/main.py (hunk 4 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py:28]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 5 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py:38]: done

File: lab_1_keywords_tfidf/main.py (hunk 1 above)
Comment (Timur20033) [reply lab_1_keywords_tfidf/main.py:167]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 2 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py:22]: done

File: lab_1_keywords_tfidf/main.py (hunk 1 above)
Comment (artyomtugaryov) [reply lab_1_keywords_tfidf/main.py:167]: do you see, changes in github? This is mean^ that it is not done

File: lab_2_keywords_cooccurrence/main.py (hunk 3 above)
Comment (artyomtugaryov) [reply lab_2_keywords_cooccurrence/main.py:28]: Why do you need this code at all?

File: lab_2_keywords_cooccurrence/main.py [hunk 6]
```diff
@@ -29,7 +57,34 @@ def extract_candidate_keyword_phrases(phrases: Sequence[str], stop_words: Sequen
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(phrases, list) or not isinstance(stop_words, list):
+        return None
+    if not all(isinstance(element, str) for element in phrases):
+        return None
+    if phrases == [] or stop_words == []:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:64]:
```suggestion
    if not phrases  or not stop_words:
```

File: lab_2_keywords_cooccurrence/main.py [hunk 7]
```diff
@@ -29,7 +57,34 @@ def extract_candidate_keyword_phrases(phrases: Sequence[str], stop_words: Sequen
...
+        for element in phrase:
+            if element not in stop_words:
+                points.append(element)
+            else:
+                points.append('.')
+        tokens = ' '.join(points)
+        back_to_string = tokens.split('.')
+        dividing = []
+        for element in back_to_string:
+            if element != '' and element != ' ':
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:81]:
```suggestion
            if element and element != ' ':
```

File: lab_2_keywords_cooccurrence/main.py [hunk 8]
```diff
@@ -55,7 +118,19 @@ def calculate_word_degrees(candidate_keyword_phrases: KeyPhrases,
...
+    if (not isinstance(candidate_keyword_phrases, list)
+            or candidate_keyword_phrases == []
+            or not isinstance(content_words, list)
+            or content_words == []):
+        return None
+    degree_dict = {}
+    for word in content_words:
+        length = 0
+        for phrase in candidate_keyword_phrases:
+            if word in phrase:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:130]:
```suggestion
            if word not in phrase:
                continue
```

File: lab_2_keywords_cooccurrence/main.py [hunk 9]
```diff
@@ -84,7 +167,19 @@ def calculate_cumulative_score_for_candidates(candidate_keyword_phrases: KeyPhra
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if (not isinstance(candidate_keyword_phrases, list) or candidate_keyword_phrases == []
+            or not isinstance(word_scores, dict) or word_scores == {}):
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:171]:
```suggestion
            or not isinstance(word_scores, dict) or not word_scores):
```

Comment (artyomtugaryov) [reply lab_2_keywords_cooccurrence/main.py:171]: Ant every time when you use `smt == {}` or `smt == []` use `NOT smth`

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — COMMENTED** [review]
Fix previos comments

File: lab_2_keywords_cooccurrence/main.py (hunk 9 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py]: understood, done

File: lab_2_keywords_cooccurrence/main.py (hunk 8 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 7 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 6 above)
Comment (Timur20033) [reply lab_2_keywords_cooccurrence/main.py]: done
